common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/RemoteBlockPushResolverSuite.java

```diff
@@ -821,20 +915,132 @@ public void testFailureAfterDuplicateBlockDoesNotInterfereActiveStream() throws
     }
   }
 
+  @Test(expected = IllegalArgumentException.class)
+  public void testBlockReceivedAfterNewAttemptRegistered()
```

Comment:
Nit: change the test name to something like `testBlockFromPreviousAttemptIsRejected`